src/components/Card.js

```diff
@@ -30,19 +43,29 @@ export default class Card {
       .addEventListener("click", () => {
         this._handleImageClick(this);
       });
-    const buttonLike = this._cardElement.querySelector(".card__button-like");
+    this.buttonLike = this._cardElement.querySelector(".card__button-like");
 
-    buttonLike.addEventListener("click", () => {
+    this.buttonLike.addEventListener("click", () => {
       this._handleLikeButton();
     });
-    const buttonTrash = this._cardElement.querySelector(".card__button-delete");
-    buttonTrash.addEventListener("click", () => {
-      this._handleTrashButton();
+    this.buttonTrash = this._cardElement.querySelector(".card__button-delete");
+    this.buttonTrash.addEventListener("click", () => {
+      //added:
+      deletePopup.open();
+
+      deletePopup.catchSelectedCard(this._cardElement, this.id);
+      //this._handleTrashButton();
     });
   }
   _handleLikeButton() {
-    const buttonLike = this._cardElement.querySelector(".card__button-like");
-    buttonLike.classList.toggle("card__button-like_active");
+    this.buttonLike.classList.toggle("card__button-like_active");
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.